File: quick_sort/quick_sort.dart
import 'dart:math';

void main() {
  var rng = new Random();
  var lista = new List<int>.generate(100, (_) => rng.nextInt(1000));
  print(quickSort(lista));
}

quickSort(List<int> list) {
  if (list.length < 2) {
    return list;
  } else {
    int pivo = list[0];
    List<int> menores = [];
    List<int> maiores = [];

    for (var i = 1; i < list.length; i++) {
      if (list[i] <= pivo) menores.add(list[i]);
    }

    for (var i = 1; i < list.length; i++) {
      if (list[i] > pivo) maiores.add(list[i]);
    }

    return [...quickSort(menores), pivo, ...quickSort(maiores)];
  }
}

File: quick_sort/quick_sort.py
import random

def quicksort(array):
    if len(array) < 2:
        return array
    else:
        pivo = array[0]
        menores = [i for i in array[1:] if i < pivo]
        maiores = [i for i in array[1:] if i > pivo]
        return quicksort(menores) + [pivo] + quicksort(maiores)

def gerar_lista():
    lista = []

    for i in range(100):
        lista.append(random.randint(0, 1000))

    return lista

print(quicksort(gerar_lista()))
